### DISCUSSION TITLE
spring cloud gateway 2.2.9 + skywalking 7.0.0

### DISCUSSION THREAD

**Discussion by G-Aaron** [apache/skywalking#8612] [Q&A] [answered]
apm-spring-cloud-gateway-2.x-plugin-7.0.0.jar  apm-spring-webflux-5.x-plugin-7.0.0.jar
 use the plugin,but the trace do not display in the correct order

**Comment by wu-sheng** [comment, marked as answer]
v7 seems too old for the community. It was years ago. 

We tested 2.2.3 for the latest v8, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/gateway-2.1.x-scenario/support-version.list#L18

**Reply by G-Aaron** [reply]
how to upgrade from v7 to v8

**Reply by wu-sheng** [reply]
It is documented for a long time, https://skywalking.apache.org/docs/main/latest/en/faq/v8-version-upgrade/#v8-upgrade

**Reply by G-Aaron** [reply]
thanks.
if i do not up, how to support for v7.

**Reply by wu-sheng** [reply]
The community doesn't cover this case, we only follow the latest.

**Reply by G-Aaron** [reply]
I want to achieve my goal by modifying the source code. Is it difficult? [https://github.com/apache/skywalking-java/blob/v7.0.0/docs/en/guides/Java-Plugin-Development-Guide.md](url)

**Reply by wu-sheng** [reply]
It is open-source, so feel free to try. As I mentioned, we don't cover this case.
